test/dynamodb/stream-query.test.js

```diff
@@ -0,0 +1,199 @@
+const {dynamodb} = require('../..');
+const sinon = require('sinon');
+const test = require('ava');
+const uuid = require('uuid/v4');
+
+const rxjs = require('rxjs');
+const { toArray, mergeMap, delay, tap } = require('rxjs/operators');
+const { useDynamoDB } = require('@lifeomic/lambda-tools').dynamodb;
+
+useDynamoDB(test, true);
+
+// 'serial' ensures this hook runs after `useDynamoDB` hooks
+test.serial.beforeEach(async (test) => {
+  const { dynamoClient } = test.context.dynamodb;
+  const table = uuid();
+
+  await dynamoClient.createTable({
+    AttributeDefinitions: [
+      { AttributeName: 'id', AttributeType: 'S' },
+      { AttributeName: 'user', AttributeType: 'S' }
+    ],
+    KeySchema: [
+      { AttributeName: 'user', KeyType: 'HASH' },
+      { AttributeName: 'id', KeyType: 'RANGE' }
+    ],
+    ProvisionedThroughput: {
+      ReadCapacityUnits: 1,
+      WriteCapacityUnits: 1
+    },
+    TableName: table
+  }).promise();
+
+  test.context.table = table;
+});
+
+function createTestStream (documentClient, table, itemsNeeded) {
+  return dynamodb.query(
+    documentClient,
+    {
+      ExpressionAttributeNames: {
+        '#user': 'user'
+      },
+      ExpressionAttributeValues: {
+        ':user': 'alice'
+      },
+      KeyConditionExpression: '#user = :user',
+      TableName: table
+    },
+    itemsNeeded
+  );
+}
+
+async function testStreamBehavior (test, initialRequestCount, startRequestCount, follupRequestCount) {
+  const { documentClient } = test.context.dynamodb;
+  const { table } = test.context;
+
+  const items = [
+    { id: 'alpha', user: 'alice' },
+    { id: 'bravo', user: 'bob' },
+    { id: 'charlie', user: 'alice' }
+  ];
+
+  const querySpy = sinon.spy(documentClient, 'query');
+
+  await documentClient.batchWrite({
```

Comment:
It's an edge case, but you might want to check `UnprocessedItems` is empty to avoid surprises

Reply:
Good call. Thanks.